pyomo/contrib/mpc/modeling/terminal.py

```diff
@@ -0,0 +1,109 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2022
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+
+from pyomo.core.base.componentuid import ComponentUID
+from pyomo.core.base.expression import Expression
+
+from pyomo.contrib.mpc.data.series_data import get_indexed_cuid
+from pyomo.contrib.mpc.data.scalar_data import ScalarData
+
+
+def get_quadratic_penalty_at_time(var, t, target, weight=None):
+    if weight is None:
+        weight = 1.0
+    return weight * (var[t] - target)**2
+
+
+def get_penalty_expressions_at_time(
+    variables,
+    time,
+    t,
+    target_data,
+    weight_data=None,
+):
+    """
+    """
```

Comment:
Missing docstring

Reply:
I've converted this function to be private and added a small description in the docstring.